lib/vmdb/plugins/automate_domain.rb

```diff
@@ -12,6 +12,10 @@ def initialize(path)
         @name            = config.fetch_path("object", "attributes", "name")
       end
 
+      def system?
+        @system ||= config.fetch_path("object", "attributes", "source") == "system"
```

Comment:
not necessarily bad, but if this evaluates to `false` it'll be evaluated next time it's called - making the memoization useless. Maybe just drop `@system`